web/src/Web.App/Services/FinanceService.cs

```diff
@@ -9,7 +12,34 @@ public interface IFinanceService
     Task<FinanceYears> GetYears();
 }
 
-public class FinanceService(IInsightApi insightApi) : IFinanceService
+public class FinanceService(
+    IInsightApi insightApi,
+    IMemoryCache memoryCache,
+    IOptions<CacheOptions> cacheOptions) : IFinanceService
 {
-    public async Task<FinanceYears> GetYears() => await insightApi.GetCurrentReturnYears().GetResultOrThrow<FinanceYears>();
-}
\ No newline at end of file
+    private readonly CacheOptions _cacheOptions = cacheOptions.Value;
+    private readonly string _cacheKey = cacheOptions.Value.CacheKey ?? throw new NullReferenceException(nameof(CacheOptions));
```

Comment:
Maybe `ArgumentNullException` instead?

Reply:
removed the exception, defaults set where values are null